docs/design/serialization.md

```diff
@@ -0,0 +1,280 @@
+Serialization
+=============
+
+Slang has a collection of serialization components. This document will be used to discuss serialization around IR/AST and modules as it currently exists. A separate document will describe the future serialization plans.
+
+All of the serialization aspects here focus on binary serialization. 
+
+The major components are
+
+* IR Serialization
+* AST/Generalized Serialization
+* SourceLoc Serialization
+* Riff container
+* C++ Extractor
+
+Generalized Serialization
+=========================
+
+Generalized serializtion is the mechanism used to save 'arbitray' C++ structures. It is currently used for serializing the AST. Although not necessary, generalized serialization is typically helped out by the `C++ extractor`, which can rudamentarily parse C++ source, and extract class-like types and their fields. The extraction then produces header files that contain macros that can then be used to drive serialization. 
```

Comment:
"serializtion" (Probably worth running a spell-check on the whole thing since I'm unlikely to catch everything)